### PR TITLE
Add CSI driver operator docs

### PR DESCRIPTION
**Pull request by jsafrane**
Add docs about adding new CSI driver using CSI driver operator & library-go.

### CODE REVIEW

File: docs/container-storage-interface-driver/index.md [hunk 1]
```diff
@@ -1,3 +1,185 @@
 # Container Storage Interface Driver
 
-TODO
+## Goal
+
+In the end, OpenShift installation should provide a cluster that is usable out of the box. Users can create PVCs or StatefulSets and get a decent storage for them.
+
+* Install one CSI driver, the most useful for the platform. Usually it’s one that provides block volumes (AWS EBS, GCP PD, Azure Disk, OpenStack Cinder, ...).
+* Provide *one* default StorageClass that is good for a generic usage. Not too expensive, but not too slow either.
+  * Each cluster is different, don’t anticipate application needs and leave creation of additional StorageClasses to the cluster administrators. They know better what they need.
```
Comment (elmiko) [inline docs/container-storage-interface-driver/index.md:9]: is this advocating that the reader should not anticipate the applications needs?

if so, i would suggest this change:
```suggestion
  * Each cluster is different, don’t anticipate application needs, leave creation of additional StorageClasses to the cluster administrators. They know better what they need.
```

File: docs/container-storage-interface-driver/index.md [hunk 2]
```diff
@@ -1,3 +1,185 @@
...
+
+OpenShift shall handle TLS side of things, i.e. it will provide [a Secret with TLS key](https://github.com/openshift/gcp-pd-csi-driver-operator/blob/223a251c3ba39d8af605258d14794b32a5cfafda/assets/service.yaml#L5) that can be used in [`kube-rbac-proxy` sidecars](https://github.com/openshift/gcp-pd-csi-driver-operator/blob/223a251c3ba39d8af605258d14794b32a5cfafda/assets/controller.yaml#L114-L115) using [Secret volumes](https://github.com/openshift/gcp-pd-csi-driver-operator/blob/223a251c3ba39d8af605258d14794b32a5cfafda/assets/controller.yaml#L127-L129). 
+
+### CI
+If you do not do so yet, test your CSi driver in vanilla Kubernetes frequently using [Kubernetes storage tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/external). No OpenShift nor operator is needed here and all your Kubernetes customers will benefit from it.
```
Comment (elmiko) [inline docs/container-storage-interface-driver/index.md:90]: minor nit
```suggestion
If you do not do so yet, test your CSI driver in vanilla Kubernetes frequently using [Kubernetes storage tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/external). Neither OpenShift or an operator is needed here and all your Kubernetes customers will benefit from it.
```

File: docs/container-storage-interface-driver/index.md [hunk 3]
```diff
@@ -1,3 +1,185 @@
...
+    ```
+
+In the end, a CI job will run the same tests. CI for a CSI driver is tricky to set up and probably should be done by Red Hat.
+
+## Deliverables
+
+### CSI driver
+Fork the CSI driver under `github.com/openshift` and integrate it with Prow and Tide as described [elsewhere in this guide](../procedures).
+
+Make sure that the CSI driver image is built in our CI.
```
Comment (elmiko) [inline docs/container-storage-interface-driver/index.md:107]: minor nit
```suggestion
Make sure that the CSI driver image is built in Red Hat's CI.
```

File: docs/container-storage-interface-driver/index.md [hunk 4]
```diff
@@ -1,3 +1,185 @@
...
+
+
+### openshift/api changes
+Add your CSI driver name to allowed `ClusterCSIDriver` names [here](https://github.com/openshift/api/blob/b632c5fc10c0ea86cb18577db9388109a43d465f/operator/v1/types_csi_cluster_driver.go#L44-L57) and [here](https://github.com/openshift/api/blob/master/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml-patch). You must update `cluster-storage-operator` to the new `openshift/api` version to get the `ClusterCSIDriver` CRD updated during cluster installation / update. A simple `go get -u github.com/openshift/api@master` is enough.
+
+### CSI driver operator
+The CSI driver must be installed via an operator and the operator must be based on library-go. As listed above, it has a lot of knowledge about OpenShift. **You cannot use an operator previously developed in-house and using OLM!**
+
+* Get a new empty repo in `gihub.com/openshift`. Take [the GCP PD CSI driver operator](https://github.com/openshift/gcp-pd-csi-driver-operator) as a base and copy it there. Replace traces of "GCP PD" and "gcp-pd" in the whole repo with your CSI driver name. If you're lucky, you may be done!
```
Comment (elmiko) [inline docs/container-storage-interface-driver/index.md:116]: i think it is worth mentioning that the red hat contact will need to make the repo, and we can link to the [procedure doc](https://github.com/elmiko/ocp-infrastructure-provider-onboarding-guide/blob/main/docs/procedures/creating-an-openshift-repository.md)

**Review by elmiko — COMMENTED** [review]
i think this generally looks awesome! thanks so much for the detailed and thoughtful docs @jsafrane . 

i have a couple minor nits/questions

**Comment by jsafrane** [comment]
@elmiko thanks for all your suggestions, I accepted them. And added a note about Red Hat creating the operator repo.

**Comment by elmiko** [comment]
thanks @jsafrane !

i'd like to cleanup a couple other minor nits i noticed while re-reading, but i don't think we need to hold up this pr to make it happen.

**Pull request merged**